homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -194,15 +195,25 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
                             .insert(rendezvous_node, cookie);
 
                         // dial discovered peers
-                        for registration in registrations {
+                        for (index, registration) in registrations.iter().enumerate() {
                             // TODO: do anything with ttl here?
                             let opts = DialOpts::peer_id(registration.record.peer_id())
                                 .addresses(registration.record.addresses().to_vec())
                                 .condition(libp2p::swarm::dial_opts::PeerCondition::Disconnected)
                                 .build();
-                            // TODO: we might be dialing too many peers here. Add settings to configure when we stop dialing new peers
-                            if let Err(err) = event_handler.swarm.dial(opts) {
-                                warn!(peer_id=registration.record.peer_id().to_string(), err=?err, "failed to dial peer discovered through rendezvous")
+
+                            // Dial discovered peer if not at connected peers limit
+                            if event_handler.connected_peers.len() + index
```

Comment:
minor: we can calculate the len() outside the loop, right?

Reply:
Also, if the connected_peers list is already > than the limit, then we don't even need the loop.

Reply:
We want to calculate inside the loop because we want to dial some discovered peers, but not all of them. 

For example, say our `connected_peers_limit` is 50 and we are currently connected to 40 peers. We receive a set of registrations that includes 20 peers. We can loop and dial the first 10 discovered peers, and ignore the remaining 10 peers.

Thinking about this some more, we can probably do better. We could add a check to not dial peers that we are already connected to. That way, we can maximize the set of registrations to establish the most connections.

Reply:
Think we still want some of the other pre-checks, but I like the last idea to avoid the extra dials.

Reply:
Updated to only dial peers we aren't already connected to. 

Had a moment when I thought we might be re-implementing the benefits we get from rendezvous cookies, but I think this is still valuable because we might dial peers outside of a rendezvous discovery context.